libsql-storage-server/src/store.rs

```diff
@@ -0,0 +1,19 @@
+use async_trait::async_trait;
+use bytes::Bytes;
+
+#[async_trait]
+pub trait FrameStore: Send + Sync {
+    async fn insert_frame(&mut self, namespace: &str, page_no: u64, frame: bytes::Bytes) -> u64;
+    async fn insert_frames(&mut self, namespace: &str, frames: Vec<FrameData>) -> u64;
```

Comment:
imo these should all take &self and handle locking internally.

Could we combine these two trait methods together?

Reply:
> imo these should all take &self and handle locking internally.

right. I was planning to remove that in the future PRs, but I did it now. 

> Could we combine these two trait methods together?

I need them both for now. In the future versions, I will remove `insert_frame`.